www/class/centreonHost.class.php

```diff
@@ -114,6 +117,80 @@ public function getList($enable = false, $template = false, $exclude = null)
         return $listHost;
     }
 
+    /**
+     * get the template currently saved for this host
+     *
+     * @param $hostId
+     *
+     * @return array
+     */
+    public function getSavedTpl($hostId): array
+    {
+        $mTp = [];
+        $dbResult = $this->db->prepare(
+            'SELECT host_tpl_id, host.host_name
+            FROM host_template_relation, host
+            WHERE host_host_id = :hostId
+            AND host_tpl_id = host.host_id'
+        );
+        $dbResult->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+        $dbResult->execute();
+        while ($multiTp = $dbResult->fetch()) {
+            $mTp[$multiTp["host_tpl_id"]] = $multiTp["host_name"];
+        }
+
+        return $mTp;
+    }
+
+    /**
+     *  get number of hosts
+     *
+     * @return int
+     */
+    private function getHostNumber(): int
+    {
+        $query = $this->db->query('SELECT COUNT(*) AS `num` FROM host WHERE host_register <> "0"');
```

Comment:
It is necessary to be stricter, select directly the right type
```suggestion
        $query = $this->db->query('SELECT COUNT(*) AS `num` FROM host WHERE host_register = "1"');
```

Reply:
shouldn't we consider the AD hosts too ?